.github/workflows/icu_ci.yml

```diff
@@ -498,18 +498,97 @@ jobs:
       - run: mvn -f tools/release/java/pom.xml package dependency:analyze
 
   # Run unit tests with UCONFIG_NO_XXX variations.
-  uconfig_variation-check-unit-tests:
+  uconfig-unit-tests:
     runs-on: ubuntu-latest
+    strategy:
+      # "fail-fast: false" let other jobs keep running even if the test break in some other uconfig.
+      fail-fast: false
+      matrix:
+        uconfig_cppflags:
+         # Ignore the following two.
+         # - "-DUCONFIG_NO_FILE_IO=1"
+         # - "-DUCONFIG_NO_CONVERSION=1"
+         - "-DUCONFIG_NO_LEGACY_CONVERSION=1"
+         - "-DUCONFIG_NO_NORMALIZATION=1"
+         - "-DUCONFIG_NO_BREAK_ITERATION=1"
+         - "-DUCONFIG_NO_IDNA=1"
+         - "-DUCONFIG_NO_COLLATION=1"
+         - "-DUCONFIG_NO_FORMATTING=1"
+         - "-DUCONFIG_NO_TRANSLITERATION=1"
+         - "-DUCONFIG_NO_REGULAR_EXPRESSIONS=1"
+         - "-DUCONFIG_NO_SERVICN=1"
+         - "-DUCONFIG_NO_FILTERED_BREAK_ITERATION=1"
+         # Turn on all the define in one test
+         - "-DUCONFIG_NO_LEGACY_CONVERSION=1 -DUCONFIG_NO_NORMALIZATION=1 -DUCONFIG_NO_BREAK_ITERATION=1 -DUCONFIG_NO_IDNA=1 -DUCONFIG_NO_COLLATION=1 -DUCONFIG_NO_FORMATTING=1 -DUCONFIG_NO_TRANSLITERATION=1 -DUCONFIG_NO_REGULAR_EXPRESSIONS=1 -DUCONFIG_NO_SERVICN=1 -DUCONFIG_NO_FILTERED_BREAK_ITERATION=1"
     steps:
-      - uses: actions/checkout@v2
-      - run: python3 tools/scripts/uconfig_vars_test.py -u
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Verify no additional new UCONFIG_NO_xxx added
+        run: |
+          # Test we have exact 12 "#ifndef UCONFIG_NO_" in uconfig.h. If the number change, we need to also
+          # adjust the uconfig_cppflags above to include the new one and update the "12" below.
+          if [ "12" != `egrep "#ifndef UCONFIG_NO_" icu4c/source/common/unicode/uconfig.h| wc -l` ]; then
+              echo "More than %s UCONFIG_NO_* defined in uconfig.h, please adjust uconfig_cppflags above to include newly added flag";
+              echo "Currently UCONFIG_NO_* defined in uconfig.h:";
+              egrep "#ifndef UCONFIG_NO_" icu4c/source/common/unicode/uconfig.h;
+              exit -1
+          fi
+      - name: Build and Test
+        env:
+          CPPFLAGS: ${{ matrix.uconfig_cppflags }}
+        run: |
+          cd icu4c/source/;
+          ./runConfigureICU Linux;
+          make -j2 tests;
 
   # Run header tests with UCONFIG_NO_XXX variations.
-  uconfig_variation-check-all-header-tests:
+  uconfig-header-tests:
```

Comment:
optional / question: This action duplicates a lot of setup from the previous uconfig-unit-tests. Would it be possible & make sense to
1. either add a second matrix variable for whether to run normal tests vs. header checks,
2. or else (nicer, simpler?) just make one action do `make -j2 tests install` and also run the header checks

?

Reply:
I tempted to do 1 but then realize it will make the steps way to complicated because they each has it's env setting and option setting on the "runConfigureICU Linux" line and cannot be shared with each other.
Suggestion 2 would neither work because the install has to be run from a  "runConfigureICU Linux" w/o the env variable.